File: src/main/java/za/ac/sun/grapl/domain/models/vertices/FileVertex.java
package za.ac.sun.grapl.domain.models.vertices;

import za.ac.sun.grapl.domain.enums.VertexBaseTraits;
import za.ac.sun.grapl.domain.enums.VertexLabels;
import za.ac.sun.grapl.domain.models.GraPLVertex;

import java.util.EnumSet;

/**
 * Node representing a source file. Often also the AST root
 */
public class FileVertex implements GraPLVertex {
    public static final VertexLabels LABEL = VertexLabels.FILE;
    public static final EnumSet<VertexBaseTraits> TRAITS = EnumSet.of(VertexBaseTraits.AST_NODE);

    public final String name;
    public final int order;

    public FileVertex(String name, int order) {
        this.name = name;
        this.order = order;
    }
}

File: src/main/java/za/ac/sun/grapl/domain/models/vertices/FileVertex.kt
package za.ac.sun.grapl.domain.models.vertices

import za.ac.sun.grapl.domain.enums.VertexBaseTraits
import za.ac.sun.grapl.domain.enums.VertexLabels
import za.ac.sun.grapl.domain.models.GraPLVertex
import java.util.*

/**
 * Node representing a source file. Often also the AST root
 */
class FileVertex(val name: String, val order: Int) : GraPLVertex {
    override fun toString(): String {
        return "FileVertex{" +
                "name='" + name + '\'' +
                ", order=" + order +
                '}'
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as FileVertex

        if (name != other.name) return false
        if (order != other.order) return false

        return true
    }

    override fun hashCode(): Int {
        var result = name.hashCode()
        result = 31 * result + order
        return result
    }

    companion object {
        @kotlin.jvm.JvmField
        val LABEL = VertexLabels.FILE
        val TRAITS: EnumSet<VertexBaseTraits> = EnumSet.of(VertexBaseTraits.AST_NODE)
    }

}